packages/zoe/src/contractFacet/zcfZygote.js

```diff
@@ -324,6 +331,9 @@ export const makeZCFZygote = async (
       instantiateIssuerStorage(issuerStorageFromZoe);
 
       const startFn = start || vivify;
+      if (privateArgsShape) {
```

Comment:
line 373, as well?

Reply:
Missed that, thanks.

Is this overall feature something you think belongs in Zoe? If so I'll make that change, clean up the `customTermsShape` handling, get tests passing, and mark it R4R.

Reply:
> Is this overall feature something you think belongs in Zoe?

Yes, it seems right for Zoe.

> If so I'll make that change, clean up the customTermsShape handling, get tests passing, and mark it R4R.

I don't understand the theory behind creating bundles and extracting artifacts well enough to be sure it'll work as is, so I'd like to see tests added showing that `WalletFactory`'s privateArgs and customTerms get enforced.

In my mental model, the shapes may need to be provided via some other mechanism than as exports from the same file, but I don't have confidence in that.